target/cheri-common/op_helper_cheri_common.c

```diff
@@ -439,21 +439,23 @@ void CHERI_HELPER_IMPL(cbuildcap(CPUArchState *env, uint32_t cd, uint32_t cb,
     } else if ((ctp->cr_uperms & cbp->cr_uperms) != ctp->cr_uperms) {
         raise_cheri_exception(env, CapEx_UserDefViolation, cb_exc);
     } else {
-        /* XXXAM basic trivial implementation may not handle
-         * compressed capabilities fully, does not perform renormalization.
+        cap_register_t result = *ctp;
+
+        /*
+         * cbuildcap is allowed to seal at any ambiently-available otype,
+         * subject to their construction conditions.
          */
-        // Without the temporary cap_register_t we would copy cb into cd
-        // if cdp cd == ct (this was caught by testing cbuildcap $c3, $c1, $c3)
-        cap_register_t result = *cbp;
-        result.cr_base = ctp->cr_base;
-        result._cr_top = ctp->_cr_top;
-        result.cr_perms = ctp->cr_perms;
-        result.cr_uperms = ctp->cr_uperms;
-        result._cr_cursor = ctp->_cr_cursor;
-        if (cap_is_sealed_entry(ctp))
-            cap_make_sealed_entry(&result);
-        else
+        if (cap_is_sealed_entry(ctp)) {
+          /* Sentries require PERM_EXECUTE in csealentry */
```

Comment:
Well they don't really require it, it's just a sanity check for "don't create something that can't be used".

Reply:
I've dropped the requirement for now, since I think it would have been unique to qemu.

Reply:
I don't think we should let CBuildCap build it if it's the only way architecturally to construct a non-executable sentry. Flute's CBuildCap doesn't currently check the permission bit for CBuildCap, only CSealEntry, but it should.

Reply:
OK.  The `sail` should probably be updated so at least our "authoritative" ISA has the right behaviour. :)